### PR TITLE
expression: implement vectorized evaluation for `builtinLowerStr`

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve? 
Implement vectorized evaluation for `builtinLowerStr`.

### What is changed and how it works?
Almost 2x faster than before:
```
BenchmarkVectorizedBuiltinFunc/builtinLowerSig-VecBuiltinFunc-12                           30000             54573 ns/op
BenchmarkVectorizedBuiltinFunc/builtinLowerSig-NonVecBuiltinFunc-12                        10000            109605 ns/op
```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: expression/builtin_string_vec.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	}
+	if types.IsBinaryStr(b.args[0].GetType()) {
+		return nil
+	}
+	for i := 0; i < input.NumRows(); i++ {
+		str := result.GetBytes(i)
+		isASCII := true
+		for _, c := range str {
+			if c >= utf8.RuneSelf {
+				isASCII = false
```
Comment (ngaut) [inline expression/builtin_string_vec.go]: add a break here?

File: expression/bench_test.go [hunk 2]
```diff
@@ -204,6 +204,9 @@ var vecExprBenchCases = map[string][]vecExprBenchCase{
 	ast.Cast: {
```
Comment (XuHuaiyu) [inline expression/bench_test.go]: Should we split vecExprBenchCases into multiple maps?
For 2 reasons:
1. Conflict will happen when a PR is ready to merge if we only define one map.
2. This map will be too long to read.

File: expression/builtin_string_vec.go (hunk 1 above)
Comment (XuHuaiyu) [inline expression/builtin_string_vec.go]: Use `continue OUTER`?

File: expression/bench_test.go (hunk 2 above)
Comment (qw4990) [reply expression/bench_test.go]: Would you have any suggestion about how to split this map？

File: expression/builtin_string_vec.go (hunk 1 above)
Comment (qw4990) [reply expression/builtin_string_vec.go]: fixed

Comment (qw4990) [reply expression/builtin_string_vec.go]: a `break` statement is added here.

Comment (qw4990) [reply expression/builtin_string_vec.go]: A `continue Outer label` is used here.

File: expression/builtin_string_vec.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package expression
+
+import (
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tidb/util/chunk"
```
Comment (zz-jason) [inline expression/builtin_string_vec.go]: separate the standard libs and the third-part libs.

File: expression/builtin_string_vec.go [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+	if err := b.args[0].VecEvalString(b.ctx, input, result); err != nil {
+		return err
+	}
+	if types.IsBinaryStr(b.args[0].GetType()) {
+		return nil
+	}
+
+Loop:
+	for i := 0; i < input.NumRows(); i++ {
+		str := result.GetBytes(i)
```
Comment (zz-jason) [inline expression/builtin_string_vec.go]: how about using `strings.ToLower()`?

**Comment by zz-jason** [comment]
to https://github.com/pingcap/tidb/issues/12058

File: expression/builtin_string_vec.go (hunk 4 above)
Comment (qw4990) [reply expression/builtin_string_vec.go]: `strings.ToLower()` will cause memory copy.

**Pull request merged**